### PR TITLE
:seedling: kustomize for e2e tests

### PR DESCRIPTION
**Pull request by khareyash05**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
use Kustomize for e2e tests

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/helper.sh [hunk 1]
```diff
@@ -66,20 +66,8 @@ function delete_cluster() {
...
-        --create-namespace \
-        --wait \
-        --set persistenceMatchNode=None \
-        --set clustersynchroManager.image.registry=localtest \
-        --set clustersynchroManager.image.repository=clustersynchro-manager-amd64 \
-        --set clustersynchroManager.image.tag=test \
-        --set apiserver.image.registry=localtest \
-        --set apiserver.image.repository=apiserver-amd64 \
-        --set apiserver.image.tag=test
+    load_image "${name}" localtest/controller-manager-amd64:test
```
Comment (Iceber) [inline test/helper.sh:69]: load_image needs to move into `create_cluster`

File: test/helper.sh [hunk 2]
```diff
@@ -66,20 +66,8 @@ function delete_cluster() {
...
+    kubectl apply -f "${ROOT}/test/kustomize"
```
Comment (Iceber) [inline test/helper.sh:70]: `kubectl kustomize "${ROOT}/test/kustomize" | kubectl apply -f -`

File: test/helper.sh [hunk 3]
```diff
@@ -66,20 +67,7 @@ function delete_cluster() {
 # install the Clusterpedia into the kind cluster
 function install_clusterpedia() {
     kubectl apply -f "${ROOT}/deploy/crds"
```
Comment (Iceber) [inline test/helper.sh:69]: kustomize will deploy the crds, so there is no need to apply them

File: test/kustomize/pvc.patch.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: internalstorage-postgres
+  namespace: clusterpedia-system
+spec:
+  select: null
```
Comment (Iceber) [inline test/kustomize/pvc.patch.yaml:7]: `selector`

**Review by Iceber — CHANGES_REQUESTED** [review]

File: test/kustomize/pvc.patch.yaml (hunk 4 above)
Comment (khareyash05) [reply test/kustomize/pvc.patch.yaml:7]: selector is not working here. its giving the issue mentioned on the slack

File: test/helper.sh [hunk 5]
```diff
@@ -54,6 +54,7 @@ function create_cluster() {
     kind create cluster --name "${name}" --image "docker.io/kindest/node:${version}"
     load_image "${name}" localtest/clustersynchro-manager-amd64:test
     load_image "${name}" localtest/apiserver-amd64:test
+    load_image "${name}" localtest/controller-manager-amd64:test
     load_image "${name}" docker.io/bitnami/postgresql:11.15.0-debian-10-r14
```
Comment (Iceber) [inline test/helper.sh:58]: https://github.com/clusterpedia-io/clusterpedia/blob/b491fe728e8105f30cd1951fd0a32cb597fa223e/kustomize/internalstorage/postgres/clusterpedia_internalstorage_deployment.yaml#L51

`load_image "${name}" postgres:12`

**Comment by Iceber** [comment]
https://github.com/clusterpedia-io/clusterpedia/blob/b491fe728e8105f30cd1951fd0a32cb597fa223e/test/test.sh#L13
No more need to check helm

**Pull request closed without merging**